// react-animate-container/dist/esm/interfaces/animations.js
export var AnimationType;
(function (AnimationType) {
    AnimationType["bounce"] = "bounce";
    AnimationType["flash"] = "flash";
    AnimationType["pulse"] = "pulse";
    AnimationType["rubberBand"] = "rubberBand";
    AnimationType["shakeX"] = "shakeX";
    AnimationType["shakeY"] = "shakeY";
    AnimationType["headShake"] = "headShake";
    AnimationType["swing"] = "swing";
    AnimationType["tada"] = "tada";
    AnimationType["wobble"] = "wobble";
    AnimationType["jello"] = "jello";
    AnimationType["heartBeat"] = "heartBeat";
    AnimationType["backInDown"] = "backInDown";
    AnimationType["backInLeft"] = "backInLeft";
    AnimationType["backInRight"] = "backInRight";
    AnimationType["backInUp"] = "backInUp";
    AnimationType["backOutDown"] = "backOutDown";
    AnimationType["backOutLeft"] = "backOutLeft";
    AnimationType["backOutRight"] = "backOutRight";
    AnimationType["backOutUp"] = "backOutUp";
    AnimationType["bounceIn"] = "bounceIn";
    AnimationType["bounceInDown"] = "bounceInDown";
    AnimationType["bounceInLeft"] = "bounceInLeft";
    AnimationType["bounceInRight"] = "bounceInRight";
    AnimationType["bounceInUp"] = "bounceInUp";
    AnimationType["bounceOut"] = "bounceOut";
    AnimationType["bounceOutDown"] = "bounceOutDown";
    AnimationType["bounceOutLeft"] = "bounceOutLeft";
    AnimationType["bounceOutRight"] = "bounceOutRight";
    AnimationType["bounceOutUp"] = "bounceOutUp";
    AnimationType["fadeIn"] = "fadeIn";
    AnimationType["fadeInDown"] = "fadeInDown";
    AnimationType["fadeInLeft"] = "fadeInLeft";
    AnimationType["fadeInRight"] = "fadeInRight";
    AnimationType["fadeInUp"] = "fadeInUp";
    AnimationType["fadeInTopLeft"] = "fadeInTopLeft";
    AnimationType["fadeInTopRight"] = "fadeInTopRight";
    AnimationType["fadeInBottomLeft"] = "fadeInBottomLeft";
    AnimationType["fadeInBottomRight"] = "fadeInBottomRight";
    AnimationType["fadeOut"] = "fadeOut";
    AnimationType["fadeOutDown"] = "fadeOutDown";
    AnimationType["fadeOutLeft"] = "fadeOutLeft";
    AnimationType["fadeOutRight"] = "fadeOutRight";
    AnimationType["fadeOutUp"] = "fadeOutUp";
    AnimationType["fadeOutTopLeft"] = "fadeOutTopLeft";
    AnimationType["fadeOutTopRight"] = "fadeOutTopRight";
    AnimationType["fadeOutBottomRight"] = "fadeOutBottomRight";
    AnimationType["fadeOutBottomLeft"] = "fadeOutBottomLeft";
    AnimationType["flip"] = "flip";
    AnimationType["flipInX"] = "flipInX";
    AnimationType["flipInY"] = "flipInY";
    AnimationType["flipOutX"] = "flipOutX";
    AnimationType["flipOutY"] = "flipOutY";
})(AnimationType || (AnimationType = {}));
